### ISSUE TITLE
CS8600 - false positive when working with nullable values that are 'Not NULL here' 

### ISSUE THREAD

**Issue by YeGaSoft** [dotnet/roslyn#63401]
(NOTE: Texts are translated to english)
(Found some similar cases that were in more specific Scenarios, but I did't found anything like this simple (common) case)

### Analyzer

**Diagnostic ID**:
[CS8600]	
"The NULL-Literal or a possible NULL-Value will be converted to Non-Nullable-Typ."
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings?f1url=%3FappId%3Droslyn%26k%3Dk(CS8600)

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.400](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

```c#
private static void CS8600Error(object? obj)
        {
            if (obj is null)
            {
                //Do Something
            }
            else
            {
                string toStr = obj.ToString();
            }
        }
```

![image](https://user-images.githubusercontent.com/72783040/184667122-17e5647a-87de-4d4c-a833-1209d2f44f13.png)

The Analyzer correctly sees that obj Is NOT NULL here 
(and when I would use obj as an Argument for a method where the parameter is not nullable there is no warning),
but when I want to call the ToString Method the error occurs
(even when the analyzer knows that the value cannot be NULL at this point)

### Expected behavior

There should be no Nullable-Warning; since the value will never be null there and the analyzer seems to know it - because it says "obj is here not null"

### Actual behavior

A warning is shown that says:
 "The NULL-Literal or a possible NULL-Value will be converted to Non-Nullable-Typ."
in the same context where the analyzer says "obj is not NULL here"

**Comment by Youssef1313** [comment]
The warning is correct.

`obj` is indeed not null. The compiler complains that `ToString` might return null, and the result is assigned to non-nullable local variable `toStr`. See also https://github.com/dotnet/runtime/issues/55509

**Comment by YeGaSoft** [comment]
@Youssef1313  Ahh, thanks for the info.
I'm gonna close the issue since it's not an analyzer-fault.

Best Regards.
